orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/NotifyDeployCompletedTask.java

```diff
@@ -40,13 +40,31 @@
 
   @Override
   public @Nonnull TaskResult executeInternal(
-      Stage stage,
-      MonitoredDeployStageData context,
-      DeploymentMonitorDefinition monitorDefinition) {
-    // TODO(mvulfson): actually populate the request data
+      Stage stage, DeploymentMonitorDefinition monitorDefinition) {
     DeploymentCompletedRequest request = new DeploymentCompletedRequest(stage);
-    monitorDefinition.getService().notifyCompleted(request);
 
+    request.setStatus(convertStageStatus((Boolean) stage.getContext().get("hasDeploymentFailed")));
+    request.setRollback(DeploymentCompletedRequest.DeploymentStatus.ROLLBACK_NOT_PERFORMED);
+
+    // check whether rollback was initiated and successful
+    if (stage.getParent() != null) {
+      stage.getParent().directChildren().stream()
+          .filter(s -> s.getType().equals(RollbackClusterStage.PIPELINE_CONFIG_TYPE))
+          .findFirst()
+          .ifPresent(
+              foundRollbackStage ->
+                  request.setRollback(
+                      convertStageStatus(
+                          foundRollbackStage.getStatus() != ExecutionStatus.SUCCEEDED)));
+    }
+
+    monitorDefinition.getService().notifyCompleted(request);
     return TaskResult.ofStatus(ExecutionStatus.SUCCEEDED);
   }
+
+  private DeploymentCompletedRequest.DeploymentStatus convertStageStatus(Boolean failedStatus) {
+    return (failedStatus == null || failedStatus)
```

Comment:
so it seems weird to me that `null` would be considered a failure. I think it should be the other way around - `null` = success. Also, the only reason you need `null` check is because of line 46... (see comment there and then you can get rid of this and make it take a little `boolean`)